test/unit/control-connection-tests.js

```diff
@@ -422,8 +422,9 @@ describe('ControlConnection', function () {
         yield this.delay;
       };
 
-      const cc = newInstance({ contactPoints: [ '::1' ], policies: { loadBalancing: lbp, reconnection: rp } },
-        getContext({ state: state, queryResults: { 'peers': [ {'rpc_address': types.InetAddress.fromString('::2') } ] }}));
+      const cc = newInstance(
+        { contactPoints: [ '::1' ], policies: { loadBalancing: lbp, reconnection: rp } },
+        getContext({ state: state, queryResults: { 'peers': [ {'rpc_address': types.InetAddress.fromString('::2') } ] }, failBorrow: [-1,1]}));
```

Comment:
A necessary change to fix one of the unit tests.  The old test relied on triggering an exception in borrowAConnection() which in turn would lead into the refresh reschedule logic.  The change that we make in this PR introduces another layer (the reresolve-then-retry logic) which now must _also_ fail in order to reach the refresh reschedule logic.  The test code overrides createConnection() with a custom implementation which uses the failBorrow array to determine whether or not it should "connect" successfully (in the context of the test) or throw an error.  Our addition causes an error to be thrown on the _second_ call to createConnection(), allowing the first to use the same logic it always has.

Interestingly the first call fails as well due to the custom data passed in via the LBP.